agent/os_helper.py

```diff
@@ -261,38 +289,23 @@ def kernel_package():
             a dict where 'source_name' and 'source_version' are the same as returned by get_packages().
         Otherwise: None
     """
-    boot_image = kernel_cmdline().get('BOOT_IMAGE')
-    if not boot_image:
+    boot_image_path = kernel_cmdline().get('BOOT_IMAGE')
+    if boot_image_path is None:
         return
-    # For apt-based distros.
-    if is_debian():
-        import apt
-
-        class FileFilter(apt.cache.Filter):
-            def apply(self, pkg):
-                return pkg.is_installed and boot_image in pkg.installed_files
-
-        cache = apt.cache.FilteredCache(apt.Cache())
-        cache.set_filter(FileFilter())
-        kernel_deb = list(cache)
-        if kernel_deb:
-            kernel_pkg = kernel_deb[0].installed
+    if is_debian():  # For apt-based distros.
+        kernel_pkg = get_kernel_deb_package(boot_image_path)
+        if kernel_pkg is not None:
             return {
-                'name': kernel_pkg.package.name,
-                'version': kernel_pkg.version,
-                'source_name': kernel_pkg.source_name,
-                'source_version': kernel_pkg.source_version,
-                'arch': kernel_pkg.architecture,
+                'name': kernel_pkg.name,
+                'version': kernel_pkg.installed.version,
+                'source_name': kernel_pkg.installed.source_name,
+                'source_version': kernel_pkg.installed.source_version,
+                'arch': kernel_pkg.installed.architecture
```

Comment:
you don't need `.installed` because `get_kernel_deb_package` already checks for that (`is_installed`).

Reply:
`.installed` here is a quite different thing